pkg/providers/vsphere/defaults.go

```diff
@@ -99,26 +101,69 @@ func (d *Defaulter) setupDefaultTemplate(ctx context.Context, spec *Spec, machin
 	return nil
 }
 
-func (d *Defaulter) setDiskDefaults(ctx context.Context, machineConfig *anywherev1.VSphereMachineConfig) error {
+func (d *Defaulter) setCloneModeDefaults(ctx context.Context, machineConfig *anywherev1.VSphereMachineConfig, datacenter string) error {
+	diskSize, err := d.govc.GetVMDiskSizeInGB(ctx, machineConfig.Spec.Template, datacenter)
+	if err != nil {
+		return fmt.Errorf("getting disk size for template %s: %v", machineConfig.Spec.Template, err)
+	}
+
 	templateHasSnapshot, err := d.govc.TemplateHasSnapshot(ctx, machineConfig.Spec.Template)
 	if err != nil {
-		return fmt.Errorf("getting template details: %v", err)
+		return fmt.Errorf("getting template snapshot details: %v", err)
 	}
 
-	if !templateHasSnapshot {
-		logger.Info("Warning: Your VM template has no snapshots. Defaulting to FullClone mode. VM provisioning might take longer.")
-		if machineConfig.Spec.DiskGiB < 20 {
-			logger.Info("Warning: VSphereMachineConfig DiskGiB cannot be less than 20. Defaulting to 20.")
-			machineConfig.Spec.DiskGiB = 20
+	// if the user didn't set the cloneMode, determine the clone mode based on the following rules:
+	// - if the template has snapshots and the template diskSize matches the machineConfig DiskGiB, default to linked clone
+	// - else use full clone
+	// if the user set the cloneMode to fullClone, use full clone
+	// if the user set the cloneMode to linkedClone, use linked clone if the following properties are met:
+	// - if the template has snapshots
+	// - if the template's disk size matches the VSphereMachineConfig diskGiB
+
+	switch machineConfig.Spec.CloneMode {
+	case "":
+		if templateHasSnapshot && machineConfig.Spec.DiskGiB == diskSize {
+			logger.V(3).Info("CloneMode not set, defaulting to linkedClone", "VSphereMachineConfig", machineConfig.Name)
+			machineConfig.Spec.CloneMode = anywherev1.LinkedClone
+		} else {
+			logger.V(3).Info("CloneMode not set, defaulting to fullClone", "VSphereMachineConfig", machineConfig.Name)
+			machineConfig.Spec.CloneMode = anywherev1.FullClone
+		}
+
+	case anywherev1.FullClone:
+		// do nothing
+
+	case anywherev1.LinkedClone:
+		if !templateHasSnapshot {
+			return fmt.Errorf(
+				"linkedClone can not be used for VSphereMachineConfig %s has the template %s has no snapshots. Ensure that the template has snapshots or change the cloneMode to fullClone",
+				machineConfig.Name,
+				machineConfig.Spec.Template,
+			)
```

Comment:
```suggestion
			return fmt.Errorf(
				"cannot use 'linkedClone' for VSphereMachineConfig (%s) because its template (%s) has no snapshots; create snapshots or change the cloneMode to 'fullClone'",
				machineConfig.Name,
				machineConfig.Spec.Template,
			)
```